### DISCUSSION TITLE
Some Questions about swift-clocks only for iOS 16 and above swift 5.7 and other for EffectTask with lib tag 0.44.1?

### DISCUSSION THREAD

**Discussion by ACFancy** [pointfreeco/swift-clocks#10] [Q&A]
- Why swift-clocks only for iOS 16 and above swift 5.7?
- Some usable functions like `EffectPublisher.timer(){} ` was marked as deprecated?
- How to migrate `Effect.merge Effect.concatenate` to `EffectTask.task(){} EffectTask.run(){} EffectTask.fireAndForget(){}`static functions? Whether is Needed to migrate?
- Is it necessary using Swift Concurrency,  `EffectTask.task(){} EffectTask.run(){} EffectTask.fireAndForget(){}` must be called in the end to return `EffectTask<Action, Never>` or `EffectTask<Action, Failure>` ?
- Because of CaseStudies  `@Dependency(\.continuousClock) `only available for  iOS 16 and above swift 5.7,
when I run in Xcode 13.4 for iOS 13 and swift 5.6, tips error.

**Comment by mbrandonw** [comment]
Hi @ACFancy, most of these questions seem to be directed at TCA rather than this repo. I will answer the one question that is swift-clocks specific, and I suggest you ask the other questions in the TCA repo.

> Why swift-clocks only for iOS 16 and above swift 5.7?

The `Clock` protocol from Apple is only available in iOS 16 and Swift 5.7+, so we have no choice but to restrict our library to those requirements.
